cmd/notation/cert/add.go

```diff
@@ -0,0 +1,81 @@
+package cert
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/notaryproject/notation/cmd/notation/internal/truststore"
+	"github.com/spf13/cobra"
+)
+
+type certAddOpts struct {
+	storeType  string
+	namedStore string
+	path       []string
+}
+
+func certAddCommand(opts *certAddOpts) *cobra.Command {
+	if opts == nil {
+		opts = &certAddOpts{}
+	}
+	command := &cobra.Command{
+		Use:   "add --type <type> --store <name> [flags] <cert_path>...",
+		Short: "Add certificates to the trust store.",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing certificate path")
+			}
+			opts.path = args
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return addCerts(opts)
+		},
+	}
+	command.Flags().StringVarP(&opts.storeType, "type", "t", "", "specify trust store type, options: ca, signingAuthority")
+	command.Flags().StringVarP(&opts.namedStore, "store", "s", "", "specify named store")
+	return command
+}
+
+func addCerts(opts *certAddOpts) error {
+	storeType := opts.storeType
+	if storeType == "" {
+		return errors.New("store type cannot be empty")
+	}
+	if !truststore.ValidateStoreType(storeType) {
+		return fmt.Errorf("unsupported store type: %s", storeType)
+	}
+	namedStore := opts.namedStore
+	if namedStore == "" {
+		return errors.New("named store cannot be empty")
```

Comment:
Do you mean values with multiple whitespaces only? The spec doesn't say we must ban such name for a named store.

Reply:
IMO we should restrict most, if not all, the user-inputs to the following regex `[a-zA-Z0-9_.-]+` i.e. alphanumeric and `_`, `-`.

This will avoid most of the [cross-platform issues](https://superuser.com/questions/358855/what-characters-are-safe-in-cross-platform-file-names-for-linux-windows-and-os)

Reply:
> IMO we should restrict most, if not all, the user-inputs to the following regex `[a-zA-Z0-9_.-]+` i.e. alphanumeric and `_`, `-`.
> 
> This will avoid most of the [cross-platform issues](https://superuser.com/questions/358855/what-characters-are-safe-in-cross-platform-file-names-for-linux-windows-and-os). Also, we can either update spec with this PR or later(please open a issue, if we choose to do it later).

The underlying [dir](https://github.com/notaryproject/notation-go/blob/bccecbc83094bffc7c942c16adcdd871bec1f940/dir/fs.go#L85) package in notation-go already utilizes the os standard library to validate the file name.

Reply:
yes but if a user is executing notation verification on multiple operating system, a trust store name like `"` would work on linux but not on windows.

Is there any reason we want to allow all possible characters? Also, not constraining input means we need to have exhaustive test and execute them on multiple operating system.

Reply:
Example:
```
➜  notation git:(truststore) ✗ color ./notation cert add /Users/pritesh/Desktop/digi-root.pem --store "/" --type ca
Successfully added following certificates to named store / of type ca:
/Users/pritesb/Desktop/digi-root.pem
➜  notation git:(truststore) ✗ color ./notation cert list                                                           
/Users/pritesb/Library/Application Support/notation/truststore/x509/ca/digi-root.pem
```

Reply:
code updated.